src/compiler/checker.ts

```diff
@@ -367,6 +367,7 @@ namespace ts {
         const globals = createSymbolTable();
         const reverseMappedCache = createMap<Type | undefined>();
         let ambientModulesCache: Symbol[] | undefined;
+        let alreadyCreatingReverseMappedType = false;

```

Comment:
This should be a state flag in `context`, instead of a local flag so it can be tracked across calls, I think.

Reply:
Done.